### PR TITLE
fix: start up the CRP controller in main.go

### PR DESCRIPTION
**Pull request by michaelawyu**
### Description of your changes

This PR fixes an issue where the CRP controller is not started up in main.go.

I have:

- [x] Run `make reviewable` to ensure this PR is ready for review.

### How has this code been tested

N/A

### Special notes for your reviewer

This PR includes some contents from #486.

### CODE REVIEW

File: cmd/hubagent/workload/setup.go [hunk 1]
```diff
@@ -119,6 +120,30 @@ func SetupControllers(ctx context.Context, mgr ctrl.Manager, config *rest.Config
...
+
+		// Start up the controller.
+		//
+		// Note that, with v1alpha1 controllers, all custom ones are started up alongside the
+		// resource change detector, when its Start() method is invoked by the controller
+		// manager. This is no longer the case with v1beta1 controllers, specifically, the
+		// v1beta1 CRP controller, as it depends on its own watchers. As a result, here
+		// this controller is spun up separately.
+		//
+		// The controller must run in a separate goroutine as Run() is a blocking call.
```
Comment (ryanzhang-oss) [inline cmd/hubagent/workload/setup.go:132]: This reconciler is added to the mgr at line 150 (old line). Is that not enough?

File: cmd/hubagent/workload/setup.go [hunk 2]
```diff
@@ -119,6 +120,30 @@ func SetupControllers(ctx context.Context, mgr ctrl.Manager, config *rest.Config
...
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+
+			err := clusterResourcePlacementControllerV1Beta1.Run(ctx, opts.ConcurrentClusterPlacementSyncs)
```
Comment (zhiying-lin) [inline cmd/hubagent/workload/setup.go:137]: can we run the controller in the changeDetector start() func, similar to v1alpha1 controller

Comment (michaelawyu) [reply cmd/hubagent/workload/setup.go]: Done, as we previously discussed.

File: cmd/hubagent/workload/setup.go (hunk 1 above)
Comment (michaelawyu) [reply cmd/hubagent/workload/setup.go]: Hi Ryan! The one at line 150 is the watcher for the v1beta1 CRP controller (it pushes keys to the workqueue); the v1beta1 CRP controller is not started at this moment (its Run() is not called).

**Review by zhiying-lin — APPROVED** [review]

**Pull request merged**